cli/app/controllers/application_controller.rb

```diff
@@ -158,6 +158,14 @@ def call(command = nil)
   #
   # @api public
   def execute
+    each_target do |target|
```

Comment:
I get the rationale for this one, it's how it was originally implemented. However, there are some edge cases, a couple of commands that don't/can't run `before_execute_on_target` and command chaining. We need to look at those cases before merging.